File: backend/routes/bugfix.py
from fastapi import APIRouter

from services.bugfix_service import BugFixService


router = APIRouter(
    prefix="/bugfix",
    tags=["Bug Fix"]
)


bugfix = BugFixService()


@router.post("/{owner}/{repo}")
def analyze_bugfix(
    owner: str,
    repo: str
):

    try:

        result = bugfix.analyze_bugfixes(
            owner,
            repo
        )

        return result

    except Exception as e:

        return {
            "error": str(e)
        }

File: frontend/src/services/bugfix.ts
const API = "http://127.0.0.1:8000";

export async function analyzeBugFixes(
  owner: string,
  repo: string
) {
  const response = await fetch(
    `${API}/bugfix/${owner}/${repo}`,
    {
      method: "POST",
    }
  );

  if (!response.ok) {
    throw new Error("Failed to analyze bugs");
  }

  return await response.json();
}